jobs/integr8ly/ocp4/install/image-deploy/Jenkinsfile

```diff
@@ -0,0 +1,175 @@
+            }
+
+            stage('Fetch and deploy operator.yaml template') {
+                sh "curl -O https://gitlab.cee.redhat.com/integreatly-qe/rhmi-jenkins/raw/master/templates/operator.yaml"
+                sh "oc create -f operator.yaml"
+            } // stage
+
+            stage ('Create Installation CR') {
+                dir('integreatly-operator') {
+                    sh(
+                        returnStdout: false,
+                        script: """
+                            sed -i "s@SELF_SIGNED_CERTS@${selfSignedCerts}@" deploy/crds/examples/integreatly-installation-cr.yaml
+                            sed -i "s@INSTALLATION_NAME@${installationName}@" deploy/crds/examples/integreatly-installation-cr.yaml
+                            sed -i "s@INSTALLATION_TYPE@${INSTALLATION_TYPE}@" deploy/crds/examples/integreatly-installation-cr.yaml
+                        """
+                    )
+
+                    sh "oc create -f deploy/crds/examples/integreatly-installation-cr.yaml -n ${installationNamespace}"
+                    sleep time: 2, unit: 'MINUTES'
+                }
+            } // stage
+
+            stage ('Verify Installation Has Finished') {
+                Map<String, Integer> stages = new LinkedHashMap<String, Integer>()
+                stages.put("bootstrap", 10)
+                stages.put("cloud-resources", 20)
+                stages.put("monitoring", 20)
+                stages.put("authentication", 20)
+                stages.put("products", 60)
+                stages.put("solution-explorer", 20)
+                String getStageStatusCommand = ""
+                
+                for (int i = 0; i < stages.size(); i++) {
+                    int counter = 0
+                    String stage = stages.keySet().toArray()[i]
+                    getStageStatusCommand = "oc get Installation ${installationName} -n ${installationNamespace} --output=json -o jsonpath='{.status.stages.${stage}.phase}'"
+                    expectedOcCommandOutput = getInstallationStatus(getStageStatusCommand)
+                    while (!expectedOcCommandOutput.contains("completed")) {
+                        println "${stage} stage has not completed yet"
+
+                        sleep time: 30, unit: 'SECONDS'
+                        expectedOcCommandOutput = getInstallationStatus(getStageStatusCommand)
+                        counter++
+                        if (counter == stages.values().toArray()[i]) {
+                            error "${stage} did not complete in time!"
+                        }
+                    }
+                    println "${stage} completed!"
+                }
+            } // stage
+        } catch (caughtError){
+            currentBuild.result = 'FAILURE'
+            err = caughtError
+        } finally {
+            notifyBuild(currentBuild.result, err)
+        }
+    }}} // timeout, ansiColor, timestamps
+} // node
+
+def getInstallationStatus(command) {
```

Comment:
This is more "an executing command" function than "getting installation status" one right :-P

Reply:
Well. Maybe to be more precise getStageStatus I think